api/api_subsystems.go

```diff
@@ -0,0 +1,63 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type MinerSubsystems []MinerSubsystem
+
+func (ms MinerSubsystems) Has(entry MinerSubsystem) bool {
+	for _, v := range ms {
+		if v == entry {
+			return true
+		}
+
+	}
+	return false
+}
+
+type MinerSubsystem int
+
+const (
+	MarketsSubsystem MinerSubsystem = iota
+	MiningSubsystem
+	SealingSubsystem
+	SectorStorageSubsystem
+)
+
+func (ms MinerSubsystem) String() string {
+	return MinerSubsystemToString[ms]
+}
+
+var MinerSubsystemToString = map[MinerSubsystem]string{
+	MarketsSubsystem:       "Markets",
+	MiningSubsystem:        "Mining",
+	SealingSubsystem:       "Sealing",
+	SectorStorageSubsystem: "SectorStorage",
+}
+
+var MinerSubsystemToID = map[string]MinerSubsystem{
+	"Markets":       MarketsSubsystem,
+	"Mining":        MiningSubsystem,
+	"Sealing":       SealingSubsystem,
+	"SectorStorage": SectorStorageSubsystem,
+}
+
+func (ms MinerSubsystem) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(MinerSubsystemToString[ms])
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
```

Comment:
Can just delegate to the json package to marshal the string: `json.Marshal(string)`. No need to write the raw output here.